src/Shared/CommunicationsUtilities.cs

```diff
@@ -336,7 +336,7 @@ internal static long GetHostHandshake(HandshakeOptions nodeType)
             //the most significant byte (leftmost 8 bits) will get zero'd out to avoid connecting to older builds.
             //| masked out | nodeHandshakeSalt | hostContext |              fileVersionHash             | SessionID
             //  0000 0000     0000 0000 0000        0000        0000 0000 0000 0000 0000 0000 0000 0000   0000 0000
-            long baseHandshake = ((long)nodeHandshakeSalt << 44) | ((long)nodeType << 40) | ((long)FileVersionHash << 8);
+            long baseHandshake = (nodeHandshakeSalt << 40) ^ ((long)nodeType << 32) ^ FileVersionHash;
```

Comment:
Remove from this PR before merging.

Reply:
I will. I wanted more validation that it worked.